meter.go

```diff
@@ -0,0 +1,230 @@
+	pgxPoolEmptyAcquire            = "pgxpool.empty_acquire"
+	pgxPoolIdleConnections         = "pgxpool.idle_connections"
+	pgxPoolMaxConnections          = "pgxpool.max_connections"
+	pgxPoolMaxIdleDestroyCount     = "pgxpool.max_idle_destroys"
+	pgxPoolMaxLifetimeDestroyCount = "pgxpool.max_lifetime_destroys"
+	pgxPoolNewConnectionsCount     = "pgxpool.new_connections"
+	pgxPoolTotalConnections        = "pgxpool.total_connections"
+)
+
+// RecordStats records database statistics for provided pgxpool.Pool at the provided interval.
+func RecordStats(db *pgxpool.Pool, opts ...StatsOption) error {
+	o := statsOptions{
+		meterProvider:              otel.GetMeterProvider(),
+		minimumReadDBStatsInterval: defaultMinimumReadDBStatsInterval,
+		defaultAttributes: []attribute.KeyValue{
+			semconv.DBSystemPostgreSQL,
+		},
+	}
+
+	for _, opt := range opts {
+		opt.applyStatsOptions(&o)
+	}
+
+	meter := o.meterProvider.Meter(meterName, metric.WithInstrumentationVersion(findOwnImportedVersion()))
+
+	return recordStats(meter, db, o.minimumReadDBStatsInterval, o.defaultAttributes...)
+}
+
+func recordStats(
+	meter metric.Meter,
+	db *pgxpool.Pool,
+	minimumReadDBStatsInterval time.Duration,
+	attrs ...attribute.KeyValue,
+) error {
+	var (
+		err error
+
+		// Asynchronous Observable Metrics
+		acquireCount            metric.Int64ObservableCounter
+		acquireDuration         metric.Float64ObservableCounter
+		acquiredConns           metric.Int64ObservableUpDownCounter
+		cancelledAcquires       metric.Int64ObservableCounter
+		constructingConns       metric.Int64ObservableUpDownCounter
+		emptyAcquires           metric.Int64ObservableCounter
+		idleConns               metric.Int64ObservableUpDownCounter
+		maxConns                metric.Int64ObservableGauge
+		maxIdleDestroyCount     metric.Int64ObservableCounter
+		maxLifetimeDestroyCount metric.Int64ObservableCounter
+		newConnsCount           metric.Int64ObservableCounter
+		totalConns              metric.Int64ObservableUpDownCounter
+
+		observeOptions             []metric.ObserveOption
+		serverAddress              = semconv.ServerAddress(db.Config().ConnConfig.Host)
+		serverPort                 = semconv.ServerPort(int(db.Config().ConnConfig.Port))
+		dbNamespace                = semconv.DBNamespace(db.Config().ConnConfig.Database)
+		poolName                   = fmt.Sprintf("%s:%d/%s", serverAddress.Value.AsString(), serverPort.Value.AsInt64(), dbNamespace.Value.AsString())
+		dbClientConnectionPoolName = semconv.DBClientConnectionPoolName(poolName)
+
+		dbStats     *pgxpool.Stat
+		lastDBStats time.Time
+
+		// lock prevents a race between batch observer and instrument registration.
+		lock sync.Mutex
+	)
+
+	lock.Lock()
+	defer lock.Unlock()
+
+	if acquireCount, err = meter.Int64ObservableCounter(
+		pgxPoolAcquireCount,
+		metric.WithDescription("Cumulative count of successful acquires from the pool."),
+	); err != nil {
+		return err
```

Comment:
Could we expand the error messages a bit?
```suggestion
		return fmt.Errorf("create %s: %w", pgxPoolAcquireCount, err) 
```

Reply:
Good call! Updated